### PR TITLE
Create OneBranch Pipeline to build and release snap packages

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

### CODE REVIEW

File: .pipelines/PowerShell-Snap-Official.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#################################################################################
+#                               OneBranch Pipelines                             #
+# This pipeline was created by EasyStart from a sample located at:              #
+#   https://aka.ms/obpipelines/easystart/samples                                #
+# Documentation:  https://aka.ms/obpipelines                                    #
+# Yaml Schema:    https://aka.ms/obpipelines/yaml/schema                        #
+# Retail Tasks:   https://aka.ms/obpipelines/tasks                              #
+# Support:        https://aka.ms/onebranchsup                                   #
+#################################################################################
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Snap-Official.yml]:
```suggestion
```

File: .pipelines/PowerShell-Snap-Official.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      - candidate
+      - beta
+      - edge
+  default: stable
+
+trigger: none
+
+variables:
+  - name: CDP_DEFINITION_BUILD_COUNT
+    value: $[counter('', 0)] # needed for onebranch.pipeline.version task https://aka.ms/obpipelines/versioning
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Snap-Official.yml]:
```suggestion
    value: $[counter('', 0)] # needed for onebranch.pipeline.version task 
```

File: .pipelines/PowerShell-Snap-Official.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  - name: LinuxContainerImage
+    value: mcr.microsoft.com/onebranch/cbl-mariner/build:2.0 # Docker image which is used to build the project https://aka.ms/obpipelines/containers
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Snap-Official.yml]:
```suggestion
    value: mcr.microsoft.com/onebranch/cbl-mariner/build:2.0 # Docker image which is used to build the project 
```

File: .pipelines/PowerShell-Snap-Official.yml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+resources:
+  repositories:
+    - repository: templates
+      type: git
+      name: OneBranch.Pipelines/GovernedTemplates
+      ref: refs/heads/main
+
+extends:
+  template: v2/OneBranch.Official.CrossPlat.yml@templates # https://aka.ms/obpipelines/templates
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Snap-Official.yml]:
```suggestion
  template: v2/OneBranch.Official.CrossPlat.yml@templates
```

File: .pipelines/PowerShell-Snap-Official.yml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  parameters:
+    git:
+      fetchDepth: 1
+      # windows only feature
+      #longpaths: true
+      retryCount: 3
+      # we don't use this and some of our agents doesn't have the feature installed
+      lfs: false
+    cloudvault: # https://aka.ms/obpipelines/cloudvault
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Snap-Official.yml]:
```suggestion
    cloudvault: 
```

File: .pipelines/PowerShell-Snap-Official.yml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    - template: /.pipelines/templates/releaseBuildAndPushStage.yml@self
+      parameters:
+        channel: preview
+        release: ${{ parameters.release }}
+
+    - template: /.pipelines/templates/releaseBuildAndPushStage.yml@self
+      parameters:
+        channel: lts
+        release: ${{ parameters.release }}
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Snap-Official.yml]:
```suggestion
        release: ${{ parameters.release }}

```

File: .pipelines/templates/releaseBuildJob.yml [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        copy-item ${env:SNAP_TARGZ} -destination $(ob_outputDirectory)/ -verbose
+      displayName: tar the snap contents from the r/w copy
+
+    # Uploads any packages as an artifact
+    - powershell: |
+        $artifactName = "drop_build_$(channel)_build_$(channel)_targz"
+        Write-Host "##vso[artifact.upload containerfolder=${artifactName};artifactname=${artifactName}]${env:SNAP_TARGZ}"
+      displayName: Publish tar.gz Artifact - $(channel)
+      condition: succeeded()
+      continueOnError: true
```
Comment (TravisEz13) [inline .pipelines/templates/releaseBuildJob.yml]:
```suggestion
      continueOnError: true

```

File: .pipelines/templates/createOutputDirectory-linux.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+steps:
+
+  - bash: |
+        mkdir -p -m a=rwx $(ob_outputDirectory)
```
Comment (adityapatwardhan) [inline .pipelines/templates/createOutputDirectory-linux.yml:4]: should we check if the folder exists before mkdir

File: .pipelines/templates/pushJob.yml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+              $releaseTag = "edge/$generatedBranch"
+          }
+          else
+          {
+              Write-verbose "non-release branch" -verbose
+              # Branch is named <previewname>
+              $releaseTag = "edge/$branchOnly"
+              $releaseTag += ([guid]::NewGuid()).ToString().Replace('-','')
+          }
+        } else {
```
Comment (adityapatwardhan) [inline .pipelines/templates/pushJob.yml]:
```suggestion
        } 
        else {
```

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**